contracts/tokenWhitelist.sol

```diff
@@ -29,15 +30,18 @@ interface ITokenWhitelist {
     function getStablecoinInfo() external view returns (string memory, uint256, uint256, bool, bool, bool, uint256);
     function tokenAddressArray() external view returns (address[] memory);
     function redeemableTokens() external view returns (address[] memory);
-    function updateTokenRate(address, uint, uint) external;
+    function methodIdWhitelist(uint32) external view returns (bool);
+    function getERC20RecipientAndAmount(bytes calldata) external view returns (address, uint);
```

Comment:
I know that it doesn't look nice, the original name was getRecipientAmount() but i changed it because in case we support more standards in the future we may need to distinguish between the different interfaces. Happy to change it though.

Reply:
OK, i think the idea of using ERC20 in the name makes sense. 

So, i think it should be : 

`getERC20RecipientAndAmount` and
`isERC20MethodSupported`

> > another thought i had, why don't we cater for the isMethodSupported function to take in the address of the ERC20 contract as a parameter. I know that as it stands we wont be using the ERC20 contract parameter, but it will allow us to extend this in the future to cater for specific functions on specific tokens. This would mean that we don't break backwards comptability in the future if we ever need this type of specialism
> 
> I thought of something similar, a method that will be able to tell if the contract is an ERC20 one an ERC721 or something else, So it gets treated accordingly in the future. Right now the functionality is very explicit but I agree that it would be nice to be able to extend it in the future. I put the ERC20 prefix because in the end this is the method that will handle only ERC20 methods but in the future we may have an ERC721 one etc.

Yes, ERC20 in the naming makes sense, but i also though the method signature should have the address of the given ERC20 passed in. Just incase we start making special cases for special Tokens. That is: 

    ```function isMethodSupported(address erc20Address, uint32 methodHash) external view returns (bool supported);```

Reply:
As for example we might support some random method only on a given token in the future.